pkg/daemon/update.go

```diff
@@ -1163,7 +1163,8 @@ func (dn *CoreOSDaemon) switchKernel(oldConfig, newConfig *mcfgv1.MachineConfig)
 		return nil
 	}
 
-	defaultKernel := []string{"kernel", "kernel-core", "kernel-modules", "kernel-modules-extra"}
+	// TODO: Drop this code and use https://github.com/coreos/rpm-ostree/issues/2542 instead
+	defaultKernel := []string{"kernel", "kernel-core", "kernel-modules", "kernel-modules-core", "kernel-modules-extra"}
```

Comment:
question: Will this work for upgrade cases too?

I think this might only work for new-install cases. For upgrade cases (e.g., someone upgrading from 4.12 to 4.13, which will imply RHCOS 8.6 -> 9.0+), we might need to do something like:

```golang
if dn.os.IsEL() && !dn.os.IsEL9() && dn.incomingOS.IsEL9() {
  // ...
}
```

Reply:
I'm aiming to fix the upgrade case i.e. 4.12 to 4.13.  Is the other case you're thinking of 4.13.x to 4.13.y?  That should also work because the relevant kernel package is present there too.

Another way to say this is: Here we're relying on the fact that this change is *only* in the 4.13 MCD, and when we're running the 4.13 MCO we know (currently) that we're going to be upgrading to a RHEL9 era OS.

Reply:
The overall code here is also currently gated on

```
	// We support Kernel update only on RHCOS and SCOS nodes
	if !dn.os.IsEL() {
		glog.Info("updating kernel on non-RHCOS nodes is not supported")
		return nil
	}
```

(There's no kernel variants in Fedora today, but if the rt kernel ever fully merges, there probably will be)

Reply:
I was thinking of the 4.13.x -> 4.13.y case because of the OS upgrade that occurs during installation. In other words: When this is installed on a new cluster, the cluster is initially brought up using the RHCOS 8.6 boot images, then eventually layers the RHCOS 9 container image onto the node. That means the MCD will run in both RHCOS 8.6 as well as RHCOS 9.

It sounds like the 4.12 -> 4.13 case is pretty clear. Although I wonder how that would work in the RT kernel case since the package names have changed.

Reply:
:tada: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3485/pull-ci-openshift-machine-config-operator-master-e2e-gcp-op/1631289321837301760/artifacts/e2e-gcp-op/test/build-log.txt has `--- PASS: TestKernelType (625.74s)`

Reply:
> I was thinking of the 4.13.x -> 4.13.y case because of the OS upgrade that occurs during installation. In other words: When this is installed on a new cluster, the cluster is initially brought up using the RHCOS 8.6 boot images, then eventually layers the RHCOS 9 container image onto the node. That means the MCD will run in both RHCOS 8.6 as well as RHCOS 9. Although this might only be an issue if someone configured their cluster on Day 1 with an RT kernel instead of doing it as a Day 2 operation.

Hmm...but that's only a transient situation for now, we're going to land the PR https://github.com/openshift/installer/pull/6886 to bump it to 9.2 as well for 4.13.

IOW, clusters born-in 4.13 will start from RHEL9 too.

But also even in the situation today, I believe what happens if you provide a "day 0" machineconfig with `kerneltype: realtime`, we'll still do the OS update and then kernel switch as a *transactional* operation (because of how rpm-ostree works - changes always operate on the pending deployment).  

IOW the MCO is doing
```
rpm-ostree rebase <rhel9>
rpm-ostree switch-to-kernel-rt
```

And even if the running host is RHEL8 (as it is today) that operation will always run targeting the RHEL9 OS.

Reply:
Got it. That makes sense!